test/integration/tc_test.go

```diff
@@ -14,20 +14,23 @@ import (
 	"github.com/stretchr/testify/require"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
 )
 
 const (
 	tcGoCounterKustomize       = "https://github.com/bpfman/bpfman/examples/config/default/go-tc-counter/?timeout=120&ref=main"
 	tcGoCounterUserspaceNs     = "go-tc-counter"
 	tcGoCounterUserspaceDsName = "go-tc-counter-ds"
+	tcGoCounterBytecodeName    = "go-tc-counter-example"
+	tcByteCodeLabelSelector    = "app.kubernetes.io/name=tcprogram"
 )
 
 func TestTcGoCounter(t *testing.T) {
 	t.Log("deploying tc counter program")
 	require.NoError(t, clusters.KustomizeDeployForCluster(ctx, env.Cluster(), tcGoCounterKustomize))
-	addCleanup(func(context.Context) error {
```

Comment:
Note: functions added via `addCleanup` are run exactly once, and that at the end of all tests. The problem with that is that this will conflict when multiple tests add the same cleanup. Instead, use t.Cleanup which runs after each test. It's also probably not necessary to throw an error on cleanup, therefore don't..?